core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java

```diff
@@ -195,11 +195,27 @@ public boolean isContainerBuild() {
     }
 
     /**
-     * The docker image to use to do the image build
+     * The docker image to use to do the image build. It can be one of `graalvm`, `mandrel`, or the full image path, e.g.
+     * {@code quay.io/quarkus/ubi-quarkus-mandrel:21.3-java17}.
      */
     @ConfigItem(defaultValue = "${platform.quarkus.native.builder-image}")
     public String builderImage;
```

Comment:
Ideally I would like this to be private to ensure all accesses go through `getBuilderImage()`.

Additionally since this setting is not expected to change while the build is running, we could save a few cycles by checking the value and setting it to point to the right image early on. Any hints on how to achieve this are welcome :)

Reply:
Unfortunately because the config objects are classes instead of interfaces, we don't have much of a choice...